### PR TITLE
Adding View all Jobs available

### PR DESCRIPTION
**Pull request by Conaill4**
includes:
- JobController,JobService,filter,view, updates to app.ts, Job Role model and validator

### CODE REVIEW

**Review by TaylorB-softdevkainos — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,20 +1,51 @@
-name: Lint Codebase
+name: Build Code Base
```
Comment (mlynch147) [inline .github/workflows/test.yml]: Wrong name for pipeline file

File: src/controllers/JobRoleController.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    try{
+        res.render("job-role-list-new.html", { JobRoles: await getJobs() } );
+    }
+    catch (e) {
+        res.locals.errormessage = e.message;
+        res.render('job-role-list-new.html');
+    }
+}
+
+export const getHomePage = async (req: express.Request, res: express.Response): Promise<void> => {
```
Comment (mlynch147) [inline src/controllers/JobRoleController.ts]: This should not be in the JobRolesController.  It should live in HomeController or something similar

File: test/ui/JobRolesTests.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { describe, it } from 'node:test';
+import webdriver from 'selenium-webdriver';
+
+describe('Job Role test', async () => {
+    it('Should view job-roles', async () => {
```
Comment (mlynch147) [inline test/ui/JobRolesTests.ts]: What is this test trying to achieve? It doesnt have any assertions?

File: test/unit/JobRoleValidatorTests.ts [hunk 4]
```diff
@@ -0,0 +1,307 @@
+import { validateJobRoles } from '../../src/validators/JobRolesValidator';
+import { JobRole } from '../../src/models/JobRole'
+import { assert, expect } from 'chai';
+import { describe, it} from 'node:test';
+
+describe('JobRolesValidator', function () {
+    describe('validateJobRoles', function () {
+      it('should not throw exception when errors', () => {
```
Comment (mlynch147) [inline test/unit/JobRoleValidatorTests.ts]: badly named....

File: test/unit/JobRoleValidatorTests.ts [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+    Write a unit test for the validateJobRoles method
+
+    When the jobRole ID number is less than 1 characters
+
+    Expect error to be returned
+
+    Should return error Invalid Id please try again
+*/
+
+it('should throw exception when jobRoleID is null', () => {
```
Comment (mlynch147) [inline test/unit/JobRoleValidatorTests.ts]: the contents of the test doesnt match the name... ??

File: test/unit/JobRoleValidatorTests.ts [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+    Write a unit test for the validateJobRoles method
+
+    When the jobRole name should be less than 5
+
+    Expect error to be returned
+
+    Should return error There is no role under that descripion
+*/
+
+it('should throw exception when rolename is less than 5 ', () => {
```
Comment (mlynch147) [inline test/unit/JobRoleValidatorTests.ts]: badly named - to me, less than 5 indicates a value of 4 or less

File: test/ui/JobRolesTests.ts
Comment (mlynch147) [inline test/ui/JobRolesTests.ts:1]: General for this file 
- indentations and formatting is wrong 
- please remove all the comments above each test... The test should be named in a way that the reader understands what should happen

File: test/unit/JobRoleValidatorTests.ts [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+
+    Write a unit test for the validateJobRoles method
+
+    When the capability greater than 6
+    Expect error to be returned
+
+    Should return error Invalid capabilityId. Please select a valid capabilityId for this job
+*/
+
+it('should throw exception when capabilityId is less than one', () => {
```
Comment (mlynch147) [inline test/unit/JobRoleValidatorTests.ts]: wrong test name

File: test/unit/JobRolesServiceTests.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+describe('JobRoleService', function () {
+    describe('getJobs', function () {
+      it('should return jobs from response', async () => {
+        const data = [jobRole];
+
+        mock.onGet(URL).reply(200, data);
+
+        const results = await getJobs();
+        expect(new Date(results[0].Date)).to.deep.equal(jobRole.Date);
```
Comment (mlynch147) [inline test/unit/JobRolesServiceTests.ts:32]: we should check the other fields as well

**Review by mlynch147 — CHANGES_REQUESTED** [review]

File: test/unit/JobRoleValidatorTests.ts (hunk 4 above)
Comment (amarauzoma) [reply test/unit/JobRoleValidatorTests.ts:8]: Martin out of curiosity, can I get more context on this. I based the naming conventions off past projects. Is there a specific way to name these test

File: test/unit/JobRolesServiceTests.ts (hunk 8 above)
Comment (amarauzoma) [reply test/unit/JobRolesServiceTests.ts:32]: For the AC on the View Job Role - https://tasks.office.com/kainos.com/en-GB/Home/Planner/#/plantaskboard?groupId=080e2153-5cc9-49af-8b98-3dc65ffde1fa&planId=IQhhunCtwUSxDbRmlZCQCJYAA0or, It's only looking for the id, role name, location, capabilityId , bandId and Date. It's extended in the view job role information ticket.

File: .github/workflows/test.yml (hunk 1 above)
Comment (amarauzoma) [reply .github/workflows/test.yml]: Have that resolved now

File: src/controllers/JobRoleController.ts (hunk 2 above)
Comment (amarauzoma) [reply src/controllers/JobRoleController.ts]: This was removed as it was code from another ticket

File: test/ui/JobRolesTests.ts
Comment (amarauzoma) [reply test/ui/JobRolesTests.ts:1]: There's no comments in JobRolesTests unless your referring to another file ?

File: test/ui/JobRolesTests.ts (hunk 3 above)
Comment (amarauzoma) [reply test/ui/JobRolesTests.ts]: Did notice this but we have made changes to that now so it should be resolved

File: test/unit/JobRoleValidatorTests.ts (hunk 4 above)
Comment (amarauzoma) [reply test/unit/JobRoleValidatorTests.ts]: Resolved

File: test/unit/JobRoleValidatorTests.ts (hunk 5 above)
Comment (amarauzoma) [reply test/unit/JobRoleValidatorTests.ts]: Updated

File: test/unit/JobRoleValidatorTests.ts (hunk 6 above)
Comment (amarauzoma) [reply test/unit/JobRoleValidatorTests.ts]: Updated

File: test/unit/JobRoleValidatorTests.ts (hunk 7 above)
Comment (amarauzoma) [reply test/unit/JobRoleValidatorTests.ts]: Changed :)

File: test/unit/JobRolesServiceTests.ts (hunk 8 above)
Comment (amarauzoma) [reply test/unit/JobRolesServiceTests.ts:32]: Updated

File: test/ui/JobRolesTests.ts
Comment (mlynch147) [reply test/ui/JobRolesTests.ts:1]: Sorry Amara - my mistake, that should be for the JobRoleValidatorTests....

**Review by mlynch147 — APPROVED** [review]

**Pull request merged**